docs/reference/internals/resolver.md

```diff
@@ -176,6 +176,38 @@ numpy==2.0.0; python_version >= "3.9" and python_version < "3.10"
 numpy==2.1.0; python_version >= "3.10"
 ```
 
+## URL dependencies
+
+In uv, a dependency can either be a registry dependency, a package with a version specifier or the
+plain package name, or a URL dependency, everything with a `{name} @ {url}` or a `git`,` url`,
+`path`, or `workspace` source, or put differently, all Git, remote file, local file or local path
+dependencies.
```

Comment:
it took me several takes to parse this bit, it could use a bit of restructuring. At the very least:
```suggestion
plain package name, or a URL dependency: everything with a `{name} @ {url}` or a `git`,` url`,
`path`, or `workspace` source, or put differently, all Git, remote file, local file or local path
dependencies.
```